node/src/service.rs

```diff
@@ -241,27 +240,46 @@ pub fn new_full(config: Configuration) -> Result<TaskManager, ServiceError> {
     let enable_grandpa = !config.disable_grandpa;
     let prometheus_registry = config.prometheus_registry().cloned();
 
-    let rpc_extensions_builder = {
+    let (rpc_extensions_builder, shared_voter_state) = {
         let client = client.clone();
-        let pool = transaction_pool.clone();
+        let transaction_pool = transaction_pool.clone();
         let select_chain = select_chain.clone();
-        let babe = {
-            crate::rpc::BabeDeps {
-                keystore: keystore_container.keystore().clone(),
-                babe_worker_handle: babe_worker_handle.clone(),
-            }
-        };
 
-        Box::new(move |deny_unsafe, _| {
-            let deps = crate::rpc::FullDeps {
-                client: client.clone(),
-                pool: pool.clone(),
-                select_chain: select_chain.clone(),
-                babe: babe.clone(),
-                deny_unsafe,
+        // BABE
+        let keystore = keystore_container.keystore();
+
+        // GRANDPA
+        let shared_voter_state_rpc = sc_consensus_grandpa::SharedVoterState::empty();
+        let shared_voter_state = shared_voter_state_rpc.clone();
+        let shared_authority_set = grandpa_link.shared_authority_set().clone();
+        let justification_stream = grandpa_link.justification_stream();
+        let finality_provider = sc_consensus_grandpa::FinalityProofProvider::new_for_service(
+            backend.clone(),
+            Some(shared_authority_set.clone()),
+        );
```

Comment:
minor aesthetic note: can you bring this outside the code block declared at line 243 ?